### PR TITLE
Phase2-hgx253 Fix error reported from static analyzer

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Fix error reported from static analyzer

#### PR validation:

Tested using runTheMatrix test workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 1]
```diff
@@ -1315,16 +1315,19 @@ std::pair<int, int> HGCalDDDConstants::waferTypeRotation(
     int layer, int waferU, int waferV, bool fromFile, bool debug) const {
   int type(HGCalTypes::WaferOut), rotn(0);
   int wl = HGCalWaferIndex::waferIndex(layer, waferU, waferV);
+  bool endOfList(true);
   if (fromFile && (waferFileSize() > 0)) {
     auto itr = hgpar_->waferInfoMap_.find(wl);
-    if (itr != hgpar_->waferInfoMap_.end()) {
+    endOfList = (itr != hgpar_->waferInfoMap_.end());
```
Comment (cvuosalo) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc:1321]: Am I reading this line correctly? `endOfList` is true if the end of the list has NOT been reached? Isn't that backwards? Am I missing something?

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by silviodonato** [comment]
+1

**Pull request merged**